src/apps/service/apps/de/job_view.clj

```diff
@@ -33,28 +37,31 @@
 
 (defn- mapped-input-subselect
   [step-id]
-  (subselect [:workflow_io_maps :wm]
-             (join [:input_output_mapping :iom] {:wm.id :iom.mapping_id})
-             (where {:iom.input      :p.id
-                     :wm.target_step step-id})))
-
-(defn- add-hidden-parameters-clause
-  [query include-hidden-params?]
-  (if-not include-hidden-params?
-    (where query {:p.is_visible true})
-    query))
+  (-> (h/select :*)
+      (h/from [:workflow_io_maps :wm])
+      (h/join [:input_output_mapping :iom] [:= :wm.id :iom.mapping_id])
+      (h/where [:and
+                [:= :iom.input :p.id]
+                [:= :wm.target_step step-id]])))
+
+(defn- get-parameters-query
+  [step-id group-id include-hidden-params?]
+  (-> (mp/hsql-params-base-query)
+      (h/order-by :p.display-order)
+      (h/where [:and
+                [:= :p.parameter_group_id group-id]
+                (when include-hidden-params? :p.is_visible)
+                [:and
+                 [:not [:exists (mapped-input-subselect step-id)]]
+                 [:or
+                  [:= :p.value_type "Input"]
+                  [:not [:coalesce :is_implicit false]]]]])
```

Comment:
I decided to switch from having two separate clauses to check for null and false values to using `coalesce` instead. It seemed a little cleaner to me.